pkg/operator/operator_test.go

```diff
@@ -217,22 +229,27 @@ func TestOperatorSyncClusterAPIControllerHealthCheckController(t *testing.T) {
 	}, {
 		featureGate:                          newFeatureGate(v1.TechPreviewNoUpgrade),
 		expectedMachineHealthCheckController: true,
+	}, {
+		featureGate:                          newFeatureGate(v1.Default),
+		expectedMachineHealthCheckController: false,
 	}}
+	infra := &v1.Infrastructure{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "cluster",
+		},
+		Status: v1.InfrastructureStatus{
+			Platform: v1.AWSPlatformType,
+		},
+	}
+	stopCh := make(<-chan struct{})
+	optr := newFakeOperator(nil, []runtime.Object{infra}, stopCh)
 
 	for _, tc := range tests {
-		infra := &v1.Infrastructure{
-			ObjectMeta: metav1.ObjectMeta{
-				Name: "cluster",
-			},
-			Status: v1.InfrastructureStatus{
-				Platform: v1.AWSPlatformType,
-			},
+		err := initFeatureGateInformer([]runtime.Object{tc.featureGate}, optr)
```

Comment:
This is still wrong. All tests in the list (`tests`) are supposed to assume clean/vanilla environment. Initializing fg informer only once leaves previous fg object in the cache. Which might (or might not) influence result of each test in the last. If you want to assume dependency between fg objects, you need to create separate unit test for it.